storage/tests/system.py

```diff
@@ -550,3 +550,91 @@ def test_rewrite_rotate_encryption_key(self):
         self.assertEqual(total, len(source_data))
 
         self.assertEqual(dest.download_as_string(), source_data)
+
+
+class TestStorageNotificationCRUD(unittest.TestCase):
+
+    topic = None
+    TOPIC_NAME = 'notification' + unique_resource_id('-')
+    CUSTOM_ATTRIBUTES = {
+        'attr1': 'value1',
+        'attr2': 'value2',
+    }
+    BLOB_NAME_PREFIX = 'blob-name-prefix/'
+
+    @property
+    def topic_path(self):
+        return 'projects/{}/topics/{}'.format(
+            Config.CLIENT.project, self.TOPIC_NAME)
+
+    def setUp(self):
+        self.case_buckets_to_delete = []
+        try:
+            from google.cloud.pubsub_v1 import PublisherClient
+        except ImportError:
+            raise unittest.SkipTest("Cannot import pubsub")
+        self.publisher_client = PublisherClient()
+        retry_429(self.publisher_client.create_topic)(self.topic_path)
+
+    def tearDown(self):
+        retry_429(self.publisher_client.delete_topic)(self.topic_path)
+        with Config.CLIENT.batch():
+            for bucket_name in self.case_buckets_to_delete:
+                bucket = Config.CLIENT.bucket(bucket_name)
+                retry_429(bucket.delete)()
```

Comment:
Does this retry make sense within the context of a batch?

Reply:
Even better:  the system test [failed at exactly this point today with a 429](https://circleci.com/gh/GoogleCloudPlatform/google-cloud-python/3430), because the `Batch.finish` can't be wrapped with a `retry_429` when invoked via a context manager.

Reply:
#3761